packages/cactus-test-geth-ledger/src/main/typescript/geth-test-ledger.ts

```diff
@@ -0,0 +1,414 @@
+import { EventEmitter } from "events";
+import Docker, { Container } from "dockerode"; // TODO - import dockerode from test-tooling?
+import { v4 as internalIpV4 } from "internal-ip";
+import Web3, { ContractAbi, TransactionReceipt } from "web3";
+import type { Web3Account } from "web3-eth-accounts";
+import { RuntimeError } from "run-time-error";
+
+import {
+  Logger,
+  Checks,
+  LogLevelDesc,
+  LoggerProvider,
+} from "@hyperledger/cactus-common";
+import { Containers } from "@hyperledger/cactus-test-tooling";
+
+export interface IGethTestLedgerOptions {
+  readonly containerImageName?: string;
+  readonly containerImageVersion?: string;
+  readonly logLevel?: LogLevelDesc;
+  readonly emitContainerLogs?: boolean;
+  readonly envVars?: string[];
+  // For test development, attach to ledger that is already running, don't spin up new one
+  readonly useRunningLedger?: boolean;
+}
+
+/**
+ * Default values used by GethTestLedger constructor.
+ */
+export const GETH_TEST_LEDGER_DEFAULT_OPTIONS = Object.freeze({
+  containerImageName: "cactus_geth_all_in_one",
+  containerImageVersion: "latest",
```

Comment:
@rwat17 Sorry, I just noticed this one: Could you please pin the image version to a specific tag that will not change in the future? It will avoid the problem of the tests breaking down later due to the floating tags.

Reply:
done